overlord/ifacestate/ifacestate.go

```diff
@@ -187,6 +189,11 @@ func connect(st *state.State, change *state.Change, plugSnap, plugName, slotSnap
 	connectPlugConnection := hookstate.HookTask(st, summary, connectPlugHookSetup, initialContext)
 	connectPlugConnection.WaitFor(connectSlotConnection)
 
+	// store IDs of the connect-plug/slot- hook tasks in the main 'connect' task,
+	// so that doConnect can conditionally mark them done on autoconnect.
+	connectInterface.Set("connect-slot-hook-task", connectSlotConnection.ID())
+	connectInterface.Set("connect-plug-hook-task", connectPlugConnection.ID())
```

Comment:
connect-slot-task and connect-plug-task might be enough for clarity here, and slightly less verbose.

Reply:
This is something I had originally and changed per @pedronis request. I think having *hook* in the name makes it clear what this is about, whereas connect-slot-task / connect-plug-task might get confused with the main connect task.
How about chaning these to "slot-hook-task" and "plug-hook-task"?

Reply:
It's okay, let's preserve the names as they are now then.